Editor/UI/Actions.cs

```diff
@@ -136,6 +131,101 @@ public static void SwitchVersion(AvailablePackage package, string version)
 				Dispatcher.Dispatch(ActionTypes.SWITCH_PACKAGE_VERSION_COMPLETE);
 			});
 		}
+
+		public static void SyncProjectEpic()
+		{
+			CompilePackageListStatus((packageListStatus) =>
+			{
+				if (!packageListStatus.IsProjectUpToDate)
+				{
+					var taskChain = new TaskChain();
+
+					foreach (var packageStatus in packageListStatus.Packages)
+					{
+						var version = packageStatus.RequiredVersion;
+						var packageName = packageStatus.PackageName;
+						var gitUrl = packageStatus.GitUrl;
+						var relativeInstallDirectory = Settings.RelativePackagesDirectoryPath + packageName;
+						var absoluteInstallDirectory = Settings.AbsolutePackagesDirectoryPath + packageName;
+
+						if (packageStatus.IsMissing)
+						{
+							taskChain.Add(
+								new AddSubmoduleTask(gitUrl, relativeInstallDirectory));
+							taskChain.Add(
+								new CheckoutSubmoduleTask(absoluteInstallDirectory, version));
+						}
+						else if (packageStatus.IsOnWrongVersion)
+						{
+							taskChain.Add(
+								new CheckoutSubmoduleTask(
+									absoluteInstallDirectory,
+									version));
+						}
+					}
+
+					taskChain.Execute(() =>
```

Comment:
Minor: you can do `taskChain.Execute(() => Debug.Log("Synced Project."));` 
I think it reads nicer